### PR TITLE
Introduce violtear

### PR DESCRIPTION
**Pull request by waghanza**
Hi @nbari,

I've added [violetear](https://violetear.org) in here.

Could you review the code base ?

I've updated results in https://github.com/waghanza/http-benchmark/blob/add_violetear/README.md#results, however their are still on **development**

Regards,

### CODE REVIEW

**Comment by waghanza** [comment]
Is the version listed right ? `go get` specify **v0.0.0** from the tag **5.0.1**

**Comment by nbari** [comment]
Hi  @waghanza thanks, the latest version is 5.0.1

File: go/violetear/main.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+func showEmpty(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "")
+}
+
+func main() {
+	router := violetear.New()
+
+	router.AddRegex(":id", `\d`)
```
Comment (nbari) [inline go/violetear/main.go]: replace `\d` with `[0-9]+` to match any number

File: FRAMEWORKS.yml [hunk 2]
```diff
@@ -247,6 +247,10 @@ go:
     github: "guregu/kami"
     version: "2.2"
     language: "1.12"
+  violetear:
+    website: violetear.org
+    version: 5.0
```
Comment (nbari) [inline FRAMEWORKS.yml]: please update version to `6.0`

**Comment by waghanza** [comment]
Hi @nbari,

Sure, I'll update. I though that for the regexp `\d` and `[0-9]` was the same
https://en.wikipedia.org/wiki/Regular_expression#Character_classes

Btw, you're right I forget the `+`.

**Comment by nbari** [comment]
@waghanza I think is safe to roll back and force, fix the regex to be `^<your regex>$` , dealing with regex is tricky and they could be matching othere handlers, for example if just using `\d` it could match:

    /users/foo123bar

When probably the desired behaviour was to match only:

    /users/123

The user could fix the regex and use `^\d+$` but that could take a while for him to find the issue.

what do you think, now that where doing a basic test?

**Comment by waghanza** [comment]
I'll then at a spec that test this case

**Comment by waghanza** [comment]
With a strict regexp `router.AddRegex(":id", `^[0-9]+$`)`, the status code given is `525`.

Should not it be 404 ?

**Comment by nbari** [comment]
Yes,  please try version  `7.0.0`,

If you use:
    
    router.AddRegex(":id", [0-9]+)
    // or
    router.AddRegex(":id", \d+)

    router.HandleFunc("/user/:id", showId, "GET")

If will return `404` for `/users/foo123bar`

and `200` for `/users/123`

File: go/violetear/main.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ func showEmpty(w http.ResponseWriter, r *http.Request) {
 func main() {
 	router := violetear.New()
 
-	router.AddRegex(":id", `[0-9]+`)
+	router.AddRegex(":id", `^[0-9]+$`)
```
Comment (nbari) [inline go/violetear/main.go]: You could indeed just use `\d+`

**Comment by waghanza** [comment]
@sebastienros How can we for `id` to be a numerical `id` in 
https://github.com/the-benchmarker/web-frameworks/blob/master/csharp/aspnetcore/Startup.cs#L23

**Comment by nbari** [comment]
@waghanza thanks for testing and your time on this 👍, just for the record using regex is quite a heavy task, indeed are mainly supported for extream cases like the one briefly described here: https://violetear.org/post/context-named/ (Duplicated named parameters):

    uuid := `[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}`
    router.AddRegex(":uuid", uuid)
    router.HandleFunc("/test/:uuid/:uuid", handleUUID, "GET,HEAD")

A catch-all alocates less, for example:

     router.HandleFunc("/test/*", handler)

And to get the param:

    func showId(w http.ResponseWriter, r *http.Request) {
	fmt.Fprintf(w, violetear.GetParam("*", r))
    }

Full example:

```go
package main

import (
	"fmt"
	"log"
	"net/http"
	"time"

	"github.com/nbari/violetear"
)

func showId(w http.ResponseWriter, r *http.Request) {
	fmt.Fprintf(w, violetear.GetParam("*", r))
}

func showEmpty(w http.ResponseWriter, r *http.Request) {
	fmt.Fprintf(w, "")
}

func main() {
	router := violetear.New()
	router.LogRequests = true

	router.AddRegex(":id", `\d+`)

	router.HandleFunc("/", showEmpty, "HEAD, GET")
	router.HandleFunc("/user/*", showId, "HEAD, GET")
	router.HandleFunc("/user", showEmpty, "POST")

	srv := &http.Server{
		Addr:           ":3000",
		Handler:        router,
		ReadTimeout:    5 * time.Second,
		WriteTimeout:   7 * time.Second,
		MaxHeaderBytes: 1 << 20,
	}
	log.Fatal(srv.ListenAndServe())
}
```

**Comment by waghanza** [comment]
@mfirry @iPomme how can we force `id` to be numerical in
https://github.com/the-benchmarker/web-frameworks/blob/master/scala/akkahttp/src/main/scala/Main.scala#L22

**Comment by mfirry** [comment]
hey @waghanza do you still need help with this?

**Pull request merged**